### ISSUE TITLE
Concierge restore removes snaps that it did not install

### ISSUE THREAD

**Issue by gruyaume** [jnsgruk/concierge#22]
## Description

Concierge's documentation claims that `concierge restore` will restore the machine to it's pre-provisioned state. However, it deletes snaps that were installed on the machine prior to concierge installing them.

For example, if the `jq` snap was installed prior to running `concierge prepare`, `concierge restore` will delete the snap regardless of that fact.

## To reproduce

1. `snap install jq`
2. `jq`
3. `sudo concierge prepare -p microk8s --microk8s-channel=1.31-strict/stable --lxd-channel "5.21/stable" --verbose`
4. `sudo concierge restore`
5. `jq`

## Expected behavior

`concierge restore` restores the machine to it's pre-provisioned state.

## Environment
- **concierge**: latest/stable:    0.9.2 2024-11-20 (21) 8MB classic

**Comment by jnsgruk** [comment]
Yes, the assumption with `concierge` has thus far been that it will be run on a pristine test machine.

I don't think it would be a big job to change this behaviour though. It probably just involves an extra check when building the `Plan` to not include snaps that are already installed.

**Comment by jnsgruk** [comment]
Closed with most recent commit, whose [commit](https://github.com/jnsgruk/concierge/commit/5f4d788f665b1240e26807d08f1eeced0a82b4a3) message read:

```
    docs: clarify functionality of restore command
    
    Previously, documentation explained the restore command as a
    command that would restore a machine to it's pre-provisioned
    state.
    
    This was not quite accurate, in that 'restore' would perform
    removal of snaps/debs/config in the plan irrespective of whether
    they happened to be in place prior to running 'prepare'.
    
    I could have changed this behaviour, but I think that would lead
    to potential inconsitencies in how `concierge` behaves depending
    on where it is run (and the machine state prior to running it),
    which somewhat defeats the purpose - so instead I have clarified
    the documentation.
    
    Closes #22.
```
